sdk/digitaltwins/Azure.DigitalTwins.Core/tests/QueryTests.cs

```diff
@@ -97,13 +99,35 @@ public async Task Query_PaginationWorks()
                 // act
                 var options = new QueryOptions();
                 options.MaxItemsPerPage = pageSize;
-                AsyncPageable<string> asyncPageableResponse = client.QueryAsync(queryString, options);
+
+                var queryWaitTimeoutStopwatch = new Stopwatch();
+                bool queryHasExpectedCount = false;
+                queryWaitTimeoutStopwatch.Start();
+                while (!queryHasExpectedCount)
+                {
+                    if (queryWaitTimeoutStopwatch.ElapsedMilliseconds >= QueryWaitTimeoutMillis)
+                    {
+                        queryWaitTimeoutStopwatch.Stop();
+                        throw new AssertionException($"Timed out waiting for at least {pageSize + 1} twins to be queryable");
+                    }
+
+                    AsyncPageable<string> asyncPageableResponse = client.QueryAsync(queryString);
+                    int count = 0;
+                    await foreach (string queriedTwin in asyncPageableResponse)
```

Comment:
If you enumerate pages instead of items, do the pages tell you how many items are in it?

Reply:
It does. I can iterate over the pages here and just do ```count+=page.Values.Count;``` instead